api/src/test/java/org/openmrs/module/sync/SyncPersonTest.java

```diff
@@ -74,7 +75,31 @@ public void runOnParent() throws Exception {
 			}
 		});
 	}
-	
+
+	@Test
+	@Transactional(propagation = Propagation.NOT_SUPPORTED)
+	public void shouldSyncPersonWithNullDOB() throws Exception {
+		runSyncTest(new SyncTestHelper() {
+
+			public void runOnChild() throws Exception {
+				PersonService ps = Context.getPersonService();
+
+				Person person = ps.getPerson(3);
+				Assertions.assertNotNull(person.getBirthdate());
+				person.setBirthdate(null);
+				ps.savePerson(person);
+				Assertions.assertNull(person.getBirthdate());
+			}
+
+			public void runOnParent() throws Exception {
+				PersonService ps = Context.getPersonService();
+				// test to make sure that the null birthdate value has been synced
+				Person person = ps.getPerson(3);
+				Assertions.assertNull(person.getBirthdate());
+			}
```

Comment:
@mseaton , you were right, if we change an object property to null it does not get synced. Here is a simple unit test case that confirms this suspicion. Since you refactored the sync code above, and you suspected this was a problem, do you mind fixing this problem with null values too? Thanks!

Reply:
@cioan - it was not a simple fix, and a pre-existing issue, so I am creating a new ticket for us to address that here:  https://issues.openmrs.org/browse/SYNC-357

Reply:
Thanks @mseaton ! This makes sense to me.

Reply:
@mseaton , please let me know if you want me to merge this PR. Thanks!

Reply:
Already merged @cioan thanks